pkg/scheduler/framework/plugins/noderesources/least_allocated_test.go

```diff
@@ -341,47 +351,52 @@ func TestNodeResourcesLeastAllocated(t *testing.T) {
 			// So the final scores are:
 			// - node1: (50 + 50 + 50) / 3 = 50
 			// - node2: (50 + 50 + 80) / 3 = 60
-			pod: &v1.Pod{Spec: cpuMemoryAndExtendedRes},
+			name: "honor extended resource if the pod requests",
+			requestedPod: &st.MakePod().Node("node1").
+				Req(map[v1.ResourceName]string{"cpu": "3000", "memory": "5000", v1.ResourceName(extendedRes): "2"}).
+				Pod,
 			nodes: []*v1.Node{
-				makeNodeWithExtendedResource("machine1", 6000, 10000, map[string]int64{extendedRes: 4}),
-				makeNodeWithExtendedResource("machine2", 6000, 10000, map[string]int64{extendedRes: 10}),
+				&st.MakeNode().Name("node1").Capacity(map[v1.ResourceName]string{"cpu": "6000", "memory": "10000", v1.ResourceName(extendedRes): "4"}).Node,
+				&st.MakeNode().Name("node2").Capacity(map[v1.ResourceName]string{"cpu": "6000", "memory": "10000", v1.ResourceName(extendedRes): "10"}).Node,
 			},
-			args:         config.NodeResourcesLeastAllocatedArgs{Resources: extendedResourceLeastAllocatedSet},
-			expectedList: []framework.NodeScore{{Name: "machine1", Score: 50}, {Name: "machine2", Score: 60}},
-			name:         "honor extended resource if the pod requests",
+			existingPods:   nil,
+			expectedScores: []framework.NodeScore{{Name: "node1", Score: 50}, {Name: "node2", Score: 60}},
+			resources:      extendedResourceLeastAllocatedSet,
 		},
 	}
 
 	for _, test := range tests {
 		t.Run(test.name, func(t *testing.T) {
-			snapshot := cache.NewSnapshot(test.pods, test.nodes)
+			state := framework.NewCycleState()
+			snapshot := cache.NewSnapshot(test.existingPods, test.nodes)
 			fh, _ := runtime.NewFramework(nil, nil, runtime.WithSnapshotSharedLister(snapshot))
-			p, err := NewLeastAllocated(&test.args, fh, feature.Features{EnablePodOverhead: true})
 
-			if test.wantErr != nil {
-				if err != nil {
-					diff := cmp.Diff(test.wantErr, err, cmpopts.IgnoreFields(field.Error{}, "BadValue", "Detail"))
-					if diff != "" {
-						t.Fatalf("got err (-want,+got):\n%s", diff)
-					}
-				} else {
-					t.Fatalf("no error produced, wanted %v", test.wantErr)
-				}
+			p, err := NewFit(
+				&config.NodeResourcesFitArgs{
+					ScoringStrategy: &config.ScoringStrategy{
+						Type:      config.LeastAllocated,
+						Resources: test.resources,
+					},
+				}, fh, plfeature.Features{EnablePodOverhead: true})
+
+			if diff := cmp.Diff(test.wantErrs.ToAggregate(), err, ignoreBadValueDetail); diff != "" {
```

Comment:
Change the logic here, I will diff these two errors whatever they are, but it `NewFit` returns with an error, I will exit  to prevent from running the code behind. @Huang-Wei